sklearn/cross_decomposition/_pls.py

```diff
@@ -494,6 +495,25 @@ def fit_transform(self, X, y=None):
         """
         return self.fit(X, y).transform(X, y)
 
+    @property
+    def coef_(self):
+        """The coefficients of the linear model."""
+        # TODO(1.3): remove and change `self._coef_` to `self.coef_`
+        #            remove catch warnings from `_get_feature_importances`
+        #            delete self._coef_no_warning
+        if getattr(self, "_coef_warning", True):
```

Comment:
I think we need to check if the attribute exist first, otherwise we would warn before the model is fitted:

```python
from sklearn.cross_decomposition import PLSRegression
X = [[0., 0., 1.], [1.,0.,0.], [2.,2.,2.], [2.,5.,4.]]
Y = [[0.1, -0.2], [0.9, 1.1], [6.2, 5.9], [11.9, 12.3]]
pls2 = PLSRegression(n_components=2)

# warning raised but no `coef_` yet
pls2.coef_
```

More specifically:

```suggestion
        if hasattr(self, "_coef_") and getattr(self, "_coef_warning", True):
```

Reply:
Nice catch